handout/a11y/key-concerns/semantic-markup.md

```diff
@@ -0,0 +1,166 @@
+        <div class="item"><a [routerLink]="['services']">Services</a></div>
+        <div class="item"><a [routerLink]="['process']">Process</a></div>
+        <div class="item"><a [routerLink]="['work']">Work</a></div>
+      </div>
+    </div>
+    <router-outlet class="page-content">
+    </router-outlet>
+  `,
+})
+```
+[View Example](http://plnkr.co/edit/cm3wBDRqIrmxpECiQhg7?p=info)
+
+While it might be obvious to someone reading the html what the purpose of each element is, or someone viewing the site, the class names don't have any semantic meaning for browsers and screen readers. We can give them more information by using the proper html elements instead.
+
+```
+@Component({
+  selector: 'ngc2-app',
+  template: `
+    <header>
+      <nav>
+        <ul>
+          <li><a [routerLink]="['']"><img src="https://angular.io/resources/images/logos/angular2/angular.svg" width="40" alt="Angular 2 logo"></a></li>
+          <li><a [routerLink]="['services']">Services</a></li>
+          <li><a [routerLink]="['process']">Process</a></li>
+          <li><a [routerLink]="['work']">Work</a></li>
+        </ul>
+      </nav>
+    </header>
+    <main aria-live="polite">
+      <router-outlet>
+      </router-outlet>
+    </main>
+  `
+})
+```
+[View Example](https://plnkr.co/edit/LHFNBsdcfbRPFnQg1DE8?p=preview)
+
+Here, we use the `header` element instead of a `div`. This lets the browser know that the elements within provide information about the site as a whole rather than about the specific page. We replace anotehr `div` with the `navigation` element. This lets the browser know the elements within are related to accessing different parts of the page or site. We also nest `router-outlet` within a `main` element, which tells the browser that the content loaded into the router outlet is the main content of the page.
+
+There are a couple of new attributes on different elements as well to give the browser even more information. The `alt` attribute has been added to the image to let the browser know that it's a logo image. There's also an `aria-live` attribute on the `main` element. This attribute is part of larger spec known as [Accessible Rich Internet Applications (WAI-ARIA)](https://www.w3.org/TR/wai-aria/) which we'll go over in detail. This is something that lets screen readers know that the content within the `main` tag will be updated on the client-side after the page has loaded and needs to be watched for updates.
+
+
+## Roles and ARIA
+
+The ARIA spec was created as a way for content authors a way to communicate more information regarding the semantics of their application rather than just details on how to render the content. This allows assistive technology to understand what's going on inside an application and relay that information in a structured and streamlined format for users with disabilities. One of the main concepts in the ARIA spec is the *role*.
```

Comment:
> The ARIA spec was created as a way for content authors ~~a way~~ to ~~communicate~~ _provide_ ~~more~~ _additional_ ~~information~~ _context_ ~~regarding~~ _to_ the semantics of their application rather than just details on how to render the content. 

Paragraph break.

> This allows assistive technology to understand what's going on inside an application and relay that information in a structured and streamlined format for users with disabilities. 

Paragraph break.

> One of the main concepts in the ARIA spec is the *role*.